docs/synchronization.md

```diff
@@ -0,0 +1,220 @@
+
+```json
+{
+  "metadata": {
+    "sync": {
+      "firstname": "Claude",
+      "lastname": "Causi",
+      "dateofbirth": "06/12/1980"
+    }
+  }
+}
+```
+
+As we cannot be sure that we will have different data, it could be better to use a custom way of providing an unique identifier.
+
+### By custom method
+
+When the external service does not provide any identifier, we have to use a custom method to generate one.
+It could be for example the filename or a hash generated with the document content.
+
+```json
+{
+  "metadata": {
+    "sync": {
+      "id:": "customhash34FED5465645ABF54656FCB"
+    }
+  }
+}
+```
+
+## Saving document based on synchronzation options
+
+cozy-konnector-libs must provide some way to easily and transparently save synchronization data. The idea is to provide additional options to the method [`saveEntry`](https://github.com/cozy/cozy-konnector-libs/blob/master/libs/saveFiles.js#L87), and by encapsulation to the method [`saveFiles`](https://github.com/cozy/cozy-konnector-libs/blob/master/libs/saveFiles.js#L149).
+
+These options are:
+
+| Option | Role |
+|--------|---------|
+| disableSyncData | (boolean) Disable storage of  synchronization data. Default: `false` |
+| findLegacyDocument | (function) Custom function which a konnector may use to look for a legacy document. Default is `null`. See below for more details.
+| getSyncData | (function) A mapping method returning additional synchronization data which have to be added to the document. |
+| idAttribute | (string&#124;Array&#124;function) How the identifier attribute is named in the entry. Default value: `id`. See below for additional customization |
+| konnector | *Mandatory* (string) The slug (or better, an uuid) of the current konnector.|
+| shouldSave | (function) A method taking current entry and matching document in database to evaluate if an entry has to be saved in database. See below default method and customization.|
+| shouldUpdate | (function) A method taking current entry and matching document in database to evaluate if a document must be updated with current entry. See below for default method and customization. |
+
+### findLegacyDocument
+If a document has not been saved yet with actual synchronization data, we will have no way to retrieve it, except by providing a `findLegacyDocument` function to `saveEntry` options. This signature of a `findLegacyDocument` function is:
+```js
+async function (entry, cozy) => { /* look for your document */ }
+```
+The `cozy` parameter is an instance of Cozy-Client.
+
+The idea is for example to retrive a document by its file path or any other accurate information.
+
+### getSyncData
+This method returns any addtional synchronization data the konnector should need. Default method should be something returning an empty object or `null`.
+```js
+(entry) => ({})
+```
+A konnector should provide its own method:
+```js
+{
+  getSyncData: (entry) => ({
+    filename: entry.name
+  })
+}
+```
+
+### idAttribute
```

Comment:
idAttributes? Since we can have several ones. What about `idEntries`? It's more meaningful for the case here to my mind.

Reply:
As we are already dealing with entities named `entries` in the konnector libs, I think it could be confusing. At first look I was not convinced with `idAttribute` either, because I did not understant well what its meaning was. It means _unique identifier attribute_ and specify the name of the entry attribute which should be use as unique identifier.